File: backend/schemas/cv.py
from typing import List, Optional
from pydantic import BaseModel


class SkillGroup(BaseModel):
    category: str
    items: List[str]


class BulletPoint(BaseModel):
    text: str


class ExperienceEntry(BaseModel):
    job_title: str
    company: str
    location: Optional[str] = None
    start_date: str
    end_date: str
    bullets: List[BulletPoint]


class EducationEntry(BaseModel):
    degree: str
    institution: str
    start_date: str
    end_date: str


class ContactInfo(BaseModel):
    name: str
    title: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    location: Optional[str] = None
    linkedin: Optional[str] = None
    portfolio: Optional[str] = None


class CVData(BaseModel):
    contact: ContactInfo
    summary: str
    skills: List[SkillGroup]
    experience: List[ExperienceEntry]
    education: List[EducationEntry]
    optimization_report: str
    match_score: int

File: frontend/src/types/cv.ts
// ─── CV Data contract ────────────────────────────────────────────────────────
// This file mirrors the Pydantic schema in backend/schemas/cv.py exactly.
// Any change to either must be reflected in both.

export interface SkillGroup {
    category: string;
    items: string[];
}


export interface BulletPoint {
    text: string;
}

export interface ExperienceEntry {
    job_title: string;
    company: string;
    location?: string;
    start_date: string;
    end_date: string;
    bullets: BulletPoint[];
}

export interface EducationEntry {
    degree: string;
    institution: string;
    start_date: string;
    end_date: string;
}

export interface ContactInfo {
    name: string;
    title?: string;
    email?: string;
    phone?: string;
    location?: string;
    linkedin?: string;
    portfolio?: string;
}

export interface CVData {
    contact: ContactInfo;
    summary: string;
    skills: SkillGroup[];
    experience: ExperienceEntry[];
    education: EducationEntry[];
    optimization_report: string;
}
